src/test/java/TicTacToe/WinLogicTest.java

```diff
@@ -0,0 +1,162 @@
+package TicTacToe;
+
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Spy;
+import org.mockito.junit.jupiter.MockitoExtension;
+
+
+@ExtendWith(MockitoExtension.class)
+public class WinLogicTest {
+    @InjectMocks
+    private WinLogic winLogic;
+    @Spy
+    private Board board;
+
+    private final char cross = 'X';
+    private final char circle = 'O';
+
+    @BeforeEach
+    public void setUp() {
+
+        board = new Board();
+        winLogic = new WinLogic(board);
```

Comment:
Nie musisz tworzyć `winLogic` w @BeforeEach możesz zrobić sobie Implementację Board w testach z ustawionym na sztywno stnaem planszy i w każdym teście robić

`@Test
public void someTest(){

//given
TestBoard testBoard =new TestBoard(List.of(new TestPostion(0, 0, Circle), new TestPostion(1, 0, Circle), )) {
winLogic = new WInLogic(testBoard )

//when
reszta testu

}
`